public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -480,6 +480,8 @@ export function filterPanelDataToQuery(data: PanelData, refId: string): PanelDat
   const error = data.error && data.error.refId === refId ? data.error : undefined;
   if (error) {
     state = LoadingState.Error;
+  } else {
+    state = LoadingState.Done;
```

Comment:
not sure this is correct, it should be water data.state 

this would set it to Done even when the state is loading 

Not sure how this is causing the crash?

Reply:
So I got this PanelData back in the componentDidUpdate
```js
{
    "state": "Error",
    "series": [
        {
            "fields": [],
            "refId": "A",
            "meta": {
                "executedQueryString": "SELECT\n  floor(extract(epoch from \"createdAt\")/60)*60 AS \"time\",\n  avg(value) as \"value\",\n  hostname as \"metric\"\nFROM \n  grafana_metr\nWHERE\n  \"createdAt\" BETWEEN '2022-02-15T17:13:05.252Z' AND '2022-02-15T18:13:05.252Z' AND\n  measurement = 'logins.count' AND\n  hostname IN('10.1.100.1','10.1.100.10','server1','server2')\nGROUP BY time, metric\nORDER BY time"
            }
        },
        {
            "refId": "B",
            "meta": {
                "type": "timeseries-wide",
                "executedQueryString": "SELECT\n  floor(extract(epoch from \"createdAt\")/60)*60 AS \"time\",\n  min(value) as \"value\",\n  'total avg' as \"metric\"\nFROM \n  grafana_metric\nWHERE\n  \"createdAt\" BETWEEN '2022-02-15T17:13:05.252Z' AND '2022-02-15T18:13:05.252Z' AND\n  measurement = 'logins.count'\nGROUP BY time\nORDER BY time"
            },
            "fields": [
                {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                        "frame": "time.Time"
                    },
                    "config": {},
                    "values": [
                        1644945180000,
                        1644945240000,
                        1644945300000,
                        1644945360000,
                        1644945420000,
                        1644945480000,
                        1644945540000,
                        1644945600000,
                        1644945660000,
                        1644945720000,
                        1644945780000,
                        1644945840000,
                        1644945900000,
                        1644945960000,
                        1644946020000,
                        1644946080000,
                        1644946140000,
                        1644946200000,
                        1644946260000,
                        1644946320000,
                        1644946380000,
                        1644946440000,
                        1644946500000,
                        1644946560000,
                        1644946620000,
                        1644946680000,
                        1644946740000,
                        1644946800000,
                        1644946860000,
                        1644946920000,
                        1644946980000,
                        1644947040000,
                        1644947100000,
                        1644947160000,
                        1644947220000,
                        1644947280000,
                        1644947340000,
                        1644947400000,
                        1644947460000,
                        1644947520000,
                        1644947580000,
                        1644947640000,
                        1644947700000,
                        1644947760000,
                        1644947820000,
                        1644947880000,
                        1644947940000,
                        1644948000000,
                        1644948060000,
                        1644948120000,
                        1644948180000,
                        1644948240000,
                        1644948300000,
                        1644948360000,
                        1644948420000,
                        1644948480000,
                        1644948540000,
                        1644948600000,
                        1644948660000,
                        1644948720000
                    ],
                    "entities": {},
                    "state": null
                },
                {
                    "name": "total avg",
                    "type": "number",
                    "typeInfo": {
                        "frame": "float64",
                        "nullable": true
                    },
                    "config": {},
                    "values": [
                        40.82163377464822,
                        43.29137773560105,
                        43.832500251947536,
                        40.74551481141019,
                        38.39422444652851,
                        34.877588583205416,
                        32.89408185225214,
                        36.24364554275727,
                        34.351136555428305,
                        33.23157438859684,
                        27.6679354698551,
                        28.60341932568494,
                        33.1316609772744,
                        37.68137197212286,
                        43.77216027957472,
                        44.53310153223197,
                        50.73323795370028,
                        46.44971793986777,
                        45.71068538075271,
                        41.61369847713669,
                        40.786400335388166,
                        48.068350285463374,
                        50.185121050247645,
                        44.665175376972414,
                        40.996063458725544,
                        41.80500800001927,
                        39.69827253678262,
                        38.053338660782956,
                        38.93544221974312,
                        37.62567439098297,
                        38.921590389938316,
                        38.959066676119235,
                        41.10268301167986,
                        43.29491664584434,
                        46.74357376004966,
                        44.54331574455377,
                        46.605699067522295,
                        38.05845851219293,
                        35.328133813946344,
                        32.85757996999067,
                        34.29506519741449,
                        32.2550339545194,
                        28.360607563009438,
                        26.56590136866933,
                        17.601510922111583,
                        19.609088070843686,
                        14.57094447477463,
                        16.258728787132007,
                        16.140953470980254,
                        13.850985800935076,
                        17.97787098432095,
                        16.991660098255206,
                        14.501857689319037,
                        15.309611226969567,
                        21.2138341767997,
                        16.831849730016923,
                        27.263789433361595,
                        33.11321942393633,
                        39.57596382359389,
                        45.028153608797844
                    ],
                    "entities": {},
                    "state": null
                }
            ],
            "length": 60
        }
    ],
    "annotations": [],
    "error": {
        "refId": "A",
        "message": "db query error: pq: relation \"grafana_metr\" does not exist"
    },
    "request": {
        "app": "dashboard",
        "requestId": "Q103",
        "timezone": "browser",
        "panelId": 2,
        "dashboardId": 27,
        "range": {
            "from": "2022-02-15T17:13:05.252Z",
            "to": "2022-02-15T18:13:05.252Z",
            "raw": {
                "from": "now-1h",
                "to": "now"
            }
        },
        "timeInfo": "",
        "interval": "1m",
        "intervalMs": 60000,
        "targets": [
            {
                "alias": "",
                "format": "time_series",
                "group": [],
                "hide": false,
                "metricColumn": "none",
                "rawQuery": true,
                "rawSql": "SELECT\n  $__timeGroup(\"createdAt\",'$summarize'),\n  avg(value) as \"value\",\n  hostname as \"metric\"\nFROM \n  grafana_metr\nWHERE\n  $__timeFilter(\"createdAt\") AND\n  measurement = 'logins.count' AND\n  hostname IN($host)\nGROUP BY time, metric\nORDER BY time",
                "refId": "A",
                "select": [
                    [
                        {
                            "params": [
                                "value"
                            ],
                            "type": "column"
                        }
                    ]
                ],
                "target": "",
                "timeColumn": "time",
                "where": [
                    {
                        "name": "$__timeFilter",
                        "params": [],
                        "type": "macro"
                    }
                ],
                "datasource": {
                    "type": "postgres",
                    "uid": "PBBCEC2D313BC06C3"
                }
            },
            {
                "alias": "",
                "format": "time_series",
                "group": [],
                "metricColumn": "none",
                "rawQuery": true,
                "rawSql": "SELECT\n  $__timeGroup(\"createdAt\",'$summarize'),\n  min(value) as \"value\",\n  'total avg' as \"metric\"\nFROM \n  grafana_metric\nWHERE\n  $__timeFilter(\"createdAt\") AND\n  measurement = 'logins.count'\nGROUP BY time\nORDER BY time",
                "refId": "B",
                "select": [
                    [
                        {
                            "params": [
                                "value"
                            ],
                            "type": "column"
                        }
                    ]
                ],
                "timeColumn": "time",
                "where": [
                    {
                        "name": "$__timeFilter",
                        "params": [],
                        "type": "macro"
                    }
                ],
                "datasource": {
                    "type": "postgres",
                    "uid": "PBBCEC2D313BC06C3"
                }
            }
        ],
        "maxDataPoints": 1470,
        "scopedVars": {
            "__interval": {
                "text": "1m",
                "value": "1m"
            },
            "__interval_ms": {
                "text": "60000",
                "value": 60000
            }
        },
        "startTime": 1644948785252,
        "rangeRaw": {
            "from": "now-1h",
            "to": "now"
        },
        "endTime": 1644948785288
    },
    "timeRange": {
        "from": "2022-02-15T17:13:05.252Z",
        "to": "2022-02-15T18:13:05.252Z",
        "raw": {
            "from": "now-1h",
            "to": "now"
        }
    },
    "timings": {
        "dataProcessingTime": 0.10000000149011612
    },
    "structureRev": 3
}
```
You can see that I have 2 queries. One failed with an error and one was successful. Because data.state is set to error for the second query that was successful it is going to send a data-error event to the view but error will be undefined.
And we don't have a check here https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/postgres/query_ctrl.ts#L319 This is where it actually crashes.